sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/V1Table.scala

```diff
@@ -31,14 +31,15 @@ import org.apache.spark.sql.types.StructType
  * An implementation of catalog v2 `Table` to expose v1 table metadata.
  */
 private[sql] case class V1Table(v1Table: CatalogTable) extends Table {
+  assert(v1Table.provider.isDefined)
```

Comment:
Why is this guaranteed? What about Hive tables, will those have provider defined?

Reply:
Only view has no provider. So this is guaranteed. Please see the parameter doc of `CatalogTable#provider`.